### PR TITLE
Support reading test-specific environment from *ENV.dhall file

### PR DESCRIPTION
**Pull request by timbertson**
This will be a new requirement for the standard test suite as part of https://github.com/dhall-lang/dhall-lang/pull/1192

I did this on my WIP branch (https://github.com/dhall-lang/dhall-haskell/pull/2236), but it seems independent enough to review early.

### CODE REVIEW

File: dhall/tests/Dhall/Test/Util.hs [hunk 1]
```diff
@@ -157,6 +167,58 @@ mockRemote url = do
...
+                where
+                  lookupHeader = liftA2 (,) (Dhall.Map.lookup "header" m) (Dhall.Map.lookup "value" m)
+                  lookupMapKey = liftA2 (,) (Dhall.Map.lookup "mapKey" m) (Dhall.Map.lookup "mapValue" m)
+          toPair _ = Nothing
+    convertEnvExpr _ = []
+
+    setEnv :: (Text, Text) -> IO (Text, Maybe Text)
+    setEnv (k, v) = do
+        old <- Turtle.need k
+        _ <- putStrLn $ Text.unpack $ "Setting env: "<>k <> "="<>v
```
Comment (sjakobi) [inline dhall/tests/Dhall/Test/Util.hs:214]: If needed, such logging should go to stderr.

**Review by sjakobi — COMMENTED** [review]
I don't see much to complain about, but I'm not very familiar with the context, so I'd prefer if @Gabriel439 could review this too.

File: dhall/tests/Dhall/Test/Util.hs (hunk 1 above)
Comment (Gabriel439) [reply dhall/tests/Dhall/Test/Util.hs:214]: I would omit the logging here

**Review by Gabriel439 — APPROVED** [review]
Looks good!  Just one comment

File: dhall/tests/Dhall/Test/Util.hs (hunk 1 above)
Comment (timbertson) [reply dhall/tests/Dhall/Test/Util.hs]: Removed 👍

**Pull request merged**